spark/src/test/scala/org/apache/comet/CometExpressionCoverageSuite.scala

```diff
@@ -0,0 +1,114 @@
+    val valuesPattern = """(?i)FROM VALUES(.+?);""".r
+    val selectPattern = """(i?)SELECT(.+?)FROM""".r
+    val builtinExamplesMap = spark.sessionState.functionRegistry
+      .listFunction()
+      .map(spark.sessionState.catalog.lookupFunctionInfo(_))
+      .filter(_.getSource.toLowerCase == "built-in")
+      .filter(f =>
+        !List("window").contains(f.getName.toLowerCase)) // exclude exotics, will run it manually
+      .map(f => {
+        val selectRows = queryPattern.findAllMatchIn(f.getExamples).map(_.group(0)).toList
+        (f.getName, selectRows.filter(_.nonEmpty))
+      })
+      .toMap
+
+    // key - function name
+    // value - list of result shows if function supported by Comet
+    val resultsMap = new mutable.HashMap[String, CoverageResult]()
+
+    builtinExamplesMap.foreach {
+      case (funcName, q :: _) =>
+        val queryResult =
+          try {
+            // Example with predefined values
+            // e.g. SELECT bit_xor(col) FROM VALUES (3), (5) AS tab(col)
+            // better option is probably to parse the query and iterate through expressions
+            // but this is adhoc coverage test
+            if (q.toLowerCase.contains(" from values")) {
+              val select = selectPattern.findFirstMatchIn(q).map(_.group(0))
+              val values = valuesPattern.findFirstMatchIn(q).map(_.group(0))
+              (select, values) match {
+                case (Some(s), Some(v)) =>
+                  testSingleLineQuery(s"select * $v", s"$s tbl")
+
+                case _ => sys.error(s"Query $q cannot be parsed properly")
+              }
+            } else {
+              // Plain example like SELECT cos(0);
+              testSingleLineQuery("select 'dummy' x", s"${q.dropRight(1)}, x from tbl")
+            }
+            CoverageResult("PASSED", Seq((q, "OK")))
+          } catch {
+            case e: TestFailedException
+                if e.message.getOrElse("").contains("Expected only Comet native operators") =>
+              CoverageResult("FAILED", Seq((q, "Unsupported")))
+            case e if e.getMessage.contains("CometNativeException") =>
+              CoverageResult("FAILED", Seq((q, "Failed on native side")))
+            case _ =>
+              CoverageResult("FAILED", Seq((q, "Failed on something else. Check query manually")))
+          }
+        resultsMap.put(funcName, queryResult)
+      case (funcName, List()) =>
+        resultsMap.put(funcName, CoverageResult("SKIPPED", Seq.empty))
+    }
+
+    // later we Convert resultMap into some HTML
+    resultsMap.toSeq.toDF("name", "details").createOrReplaceTempView("t")
+    val str_agg = showString(
+      spark.sql(
+        "select result, d._2 as reason, count(1) cnt from (select name, t.details.result, explode_outer(t.details.details) as d from t) group by 1, 2"),
+      500,
+      0)
+    Files.write(Paths.get("doc/spark_coverage_agg.txt"), str_agg.getBytes(StandardCharsets.UTF_8))
```

Comment:
```suggestion
    Files.write(Paths.get("doc/spark_agg_expr_coverage.txt"), str_agg.getBytes(StandardCharsets.UTF_8))
```

Reply:
Oh, the file is not for aggregation expression. nvm.

Reply:
But I think it may be better to have the aggregation of expression coverage in the same file?

Reply:
Filed #282